File: BAEKJOON/10610_30/10610.py
import sys

sys.stdin = open('input.txt', 'r')

number = input()


def solution(number):
    number_list = sorted(list(map(int, list(number))), reverse=True)
    if number_list[-1] == 0 and sum(number_list) % 3 == 0:
        return ''.join(map(str, number_list))
    else:
        return -1


print(solution(number))

File: BAEKJOON/10610_30/10610.js
const fs = require('fs');
const fileStream = fs.createReadStream('input.txt');
const readline = require('readline');

const rl = readline.createInterface({
  input: fileStream,
  output: process.stdout,
});

let input;

function solution(number) {
  const numberArr = number
    .split('')
    .map(Number)
    .sort((a, b) => b - a);
  if (
    numberArr[numberArr.length - 1] === 0 &&
    numberArr.reduce((acc, curr) => acc + curr, 0) % 3 === 0
  ) {
    return numberArr.join('');
  }
  return -1;
}

rl.on('line', (line) => {
  input = line;
}).on('close', () => {
  console.log(solution(input));
  process.exit();
});
